scripts/run-jenkins.sh

```diff
@@ -0,0 +1,319 @@
+#!/bin/bash -xe
+
+TIMEOUTCMD=scripts/babysitter
+export MONO_BABYSITTER_LOG_FILE=babysitter_report.json_lines
+
+if [[ ${label} == debian-arm* ]]; then export MONO_TEST_RABBITMQ_SERVER=192.168.100.12; fi
+if [[ ${label} == 'osx-i386' ]]; then export MONO_TEST_RABBITMQ_SERVER=192.168.100.12; EXTRA_CONF_FLAGS="--with-libgdiplus=/Library/Frameworks/Mono.framework/Versions/Current/lib/libgdiplus.dylib --enable-nls=no --build=i386-apple-darwin11.2.0"; fi
+if [[ ${label} == 'osx-amd64' ]]; then export MONO_TEST_RABBITMQ_SERVER=192.168.100.12; EXTRA_CONF_FLAGS="--with-libgdiplus=/Library/Frameworks/Mono.framework/Versions/Current/lib/libgdiplus.dylib --enable-nls=no"; fi
+
+STEP="configure"
+STARTTIME=`date +%s`
+echo "*** start: ${STEP}"
+${TIMEOUTCMD} --signal=ABRT --kill-after=60s 60m ./autogen.sh --with-monodroid --with-monotouch --with-monotouch_watch --with-monotouch_tv --with-xammac --with-mobile_static $EXTRA_CONF_FLAGS && echo -e "*** end($(echo $(date +%s) - ${STARTTIME} | bc)): ${STEP}: \e[42mPassed\e[0m" || (echo -e "*** end($(echo $(date +%s) - ${STARTTIME} | bc)): ${STEP}: \e[41mFailed\e[0m" && exit 1)
```

Comment:
Let's take this opportunity and factor out the "step" logic into a helper script so this becomes more readable.

E.g. something like:
```
run-step.sh --label=configure --timeout=60m ./autogen.sh .....
```